### PR TITLE
fix regression on Core.pluginDirectory when backend is builtin

### PR DESCRIPTION
**Pull request by fairclothjm**
Fix a regression on `Core.pluginDirectory`. When enabling a secrets/auth plugin, vault looks in the `Core.pluginDirectory` if the `dev-plugin-dir` flag or `plugin_directory` config is set and fails if it doesn’t find it.
 
The following error would occur when the `dev-plugin-dir` flag or `plugin_directory` config was set on a vault dev server:

```
$ vault server -dev -dev-root-token-id=root -dev-plugin-dir=$HOME/dev/plugins
$ vault version
Vault v1.11.0-dev1 ('2d03150157a76099af193264dd97cdc52c433f14')
$ vault auth enable approle
Error enabling approle auth: Error making API request.

URL: POST http://127.0.0.1:8200/v1/sys/auth/approle
Code: 400. Errors:

* error stating "/Users/jmf/dev/plugins/approle": stat /Users/jmf/dev/plugins/approle: no such file or directory
```

### CODE REVIEW

**Review by calvn — APPROVED** [review]

**Comment by calvn** [comment]
We should backport this all the way to 1.8.x as well.

**Comment by swenson** [comment]
Is there a good way to add a test for this regression?

**Comment by fairclothjm** [comment]
> Is there a good way to add a test for this regression?

@swenson I think it is a good idea. I will look into that.

**Comment by fairclothjm** [comment]
> We should backport this all the way to 1.8.x as well.

@calvn Looks like those backports were either closed or reverted.

- https://github.com/hashicorp/vault/pull/14880
- https://github.com/hashicorp/vault/pull/14914
- https://github.com/hashicorp/vault/pull/14915

**Review by swenson — APPROVED** [review]
LGTM; thanks for adding the tests!

File: vault/auth_test.go [hunk 1]
```diff
@@ -141,6 +141,33 @@ func TestCore_DefaultAuthTable(t *testing.T) {
...
+		Table: credentialTableType,
+		Path:  "approle/",
+		Type:  "approle",
+	}
+	err := c.enableCredential(namespace.RootContext(nil), me)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	match := c.router.MatchingMount(namespace.RootContext(nil), "auth/approle/bar")
```
Comment (calvn) [inline vault/auth_test.go:165]: Would the error trip in here or in `enableCredential` above? I'm wondering if this path check is necessary for the unit test.

**Review by calvn — APPROVED** [review]
One question, but otherwise looks good! Thanks for adding the tests!

File: vault/auth_test.go (hunk 1 above)
Comment (fairclothjm) [reply vault/auth_test.go]: Good point, it's not necessary

**Pull request merged**